tika-eval/tika-eval-core/src/main/java/org/apache/tika/eval/core/tokens/URLEmailNormalizingFilterFactory.java

```diff
@@ -69,11 +69,10 @@ public boolean incrementToken() throws IOException {
                 return false;
             }
             //== is actually substantially faster than .equals(String)
-            if (typeAtt.type() == UAX29URLEmailTokenizer.TOKEN_TYPES[UAX29URLEmailTokenizer.URL]) {
+            if (typeAtt.type().equals(UAX29URLEmailTokenizer.TOKEN_TYPES[UAX29URLEmailTokenizer.URL])) {
```

Comment:
This was done out of a notional sense of efficiency. I'm not sure we need to change it.

Reply:
So, parameter of the `TypeAttribute#setType` can be exactly that String (`UAX29URLEmailTokenizer.TOKEN_TYPES[UAX29URLEmailTokenizer.URL]`) ?

Reply:
This relies on the Lucene not changing the underlying static strings: https://github.com/apache/lucene/blob/main/lucene/analysis/common/src/java/org/apache/lucene/analysis/email/UAX29URLEmailTokenizer.java#L61

Reply:
OK. I think that they had to use enum instead of a string array for such a thing :blush: I'll rollback that lines at my PR.